weaver/cli.py

```diff
@@ -0,0 +1,747 @@
+        # consider possible ambiguity if literal CWL input is named 'inputs'
+        # - if value of 'inputs' is an object, it can collide with 'OGC' schema,
+        #   unless 'value/href' are present or their sub-dict don't have CWL 'class'
+        # - if value of 'inputs' is an array, it can collide with 'OLD' schema,
+        #   unless 'value/href' (and 'id' technically) are present
+        values = inputs.get("inputs", null)
+        if (
+            values is null or
+            values is not null and (
+                (isinstance(values, dict) and get_any_value(values) is null and "class" not in values) or
+                (isinstance(values, list) and all(isinstance(v, dict) and get_any_value(v) is null for v in values))
+            )
+        ):
+            values = cwl2json_input_values(inputs)
+        data = {
+            # NOTE: since sync is not yet properly implemented in Weaver, simulate with monitoring after if requested
+            # FIXME: support 'sync' (https://github.com/crim-ca/weaver/issues/247)
+            "mode": EXECUTE_MODE_ASYNC,
+            "inputs": values,
+            # FIXME: support 'response: raw' (https://github.com/crim-ca/weaver/issues/376)
+            "response": EXECUTE_RESPONSE_DOCUMENT,
+            # FIXME: allow omitting 'outputs' (https://github.com/crim-ca/weaver/issues/375)
+            # FIXME: allow 'transmissionMode: value/reference' selection (https://github.com/crim-ca/weaver/issues/377)
+            "outputs": {}
+        }
+        # FIXME: since (https://github.com/crim-ca/weaver/issues/375) not implemented, auto-populate all the outputs
+        base = self._get_url(url)
+        result = self.describe(process_id, url=base)
+        if not result.success:
+            return OperationResult(False, "Could not obtain process description for execution.",
+                                   body=result.body, headers=result.headers, code=result.code, text=result.text)
+        outputs = result.body.get("outputs")
+        for output_id in outputs:
+            # use 'value' to have all outputs reported in body as 'value/href' rather than 'Link' headers
+            data["outputs"][output_id] = {"transmissionMode": EXECUTE_TRANSMISSION_MODE_VALUE}
+
+        LOGGER.info("Executing [%s] with inputs:\n%s", process_id, _json2text(inputs))
+        path = f"{base}/processes/{process_id}/execution"  # use OGC-API compliant endpoint (not '/jobs')
+        resp = request_extra("POST", path, json=data, headers=self._headers, settings=self._settings)
+        result = self._parse_result(resp)
+        if not monitor or not result.success:
+            return result
+        # although Weaver returns "jobID" in the body for convenience,
+        # employ the "Location" header to be OGC-API compliant
+        job_url = resp.headers.get("Location", "")
+        time.sleep(1)  # small delay to ensure process execution had a chance to start before monitoring
```

Comment:
Technically, there is nothing to be robust about here. The response can only be obtained after the job was already submitted to celery. It's more to save a possible useless request that will only report "accepted" again. 
This is usually a sufficient "warm up time" for the process to validate and fail in case of error, instead of waiting for the next polling interval. For external/manual calls to monitor, the user could use any delay they see fit.

Reply:
I'm only talking about the client side. I think that the monitor should inclure the sleep rather than the execute, because if the user is to call the monitor himself the warm up would still be included. My point is really that the sleep before Weaver is ready belong to the monitoring part and not the execute. The monitor function could even detect this situation on the first loop and avoid raising an exception before retrying (that's what I meant by robustness).
I will not block this PR for that though, it's fair enough as it is.